src/main/java/it/gov/pagopa/pu/debtpositions/service/massive/DebtPositionMassiveServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class DebtPositionMassiveServiceImpl implements DebtPositionMassiveService {
+
+  private final InstallmentSynchronizeMapper installmentSynchronizeMapper;
+  private final DebtPositionRepository debtPositionRepository;
+  private final DebtPositionTypeOrgRepository debtPositionTypeOrgRepository;
+  private final DebtPositionTypeRepository debtPositionTypeRepository;
+  private final OrganizationSearchClient organizationSearchClient;
+  private final InsertionActionMassiveDebtPositionService insertionActionMassiveDebtPositionService;
+  private final UpdateActionMassiveDebtPositionService updateActionMassiveDebtPositionService;
+
+  public DebtPositionMassiveServiceImpl(InstallmentSynchronizeMapper installmentSynchronizeMapper, DebtPositionRepository debtPositionRepository, DebtPositionTypeOrgRepository debtPositionTypeOrgRepository, DebtPositionTypeRepository debtPositionTypeRepository, OrganizationSearchClient organizationSearchClient, InsertionActionMassiveDebtPositionService insertionActionMassiveDebtPositionService, UpdateActionMassiveDebtPositionService updateActionMassiveDebtPositionService) {
+    this.installmentSynchronizeMapper = installmentSynchronizeMapper;
+    this.debtPositionRepository = debtPositionRepository;
+    this.debtPositionTypeOrgRepository = debtPositionTypeOrgRepository;
+    this.debtPositionTypeRepository = debtPositionTypeRepository;
+    this.organizationSearchClient = organizationSearchClient;
+    this.insertionActionMassiveDebtPositionService = insertionActionMassiveDebtPositionService;
+    this.updateActionMassiveDebtPositionService = updateActionMassiveDebtPositionService;
+  }
+
+
+  @Override
+  public String installmentSynchronize(InstallmentSynchronizeDTO installmentSynchronizeDTO, Boolean massive, String accessToken, String operatorExternalUserId) {
+    DebtPositionDTO debtPositionSynchronizeDTO = createDpFromInstallmentSynchronize(installmentSynchronizeDTO, accessToken);
+
+    DebtPosition debtPosition = checkProcessableDebtPosition(debtPositionSynchronizeDTO.getIupdOrg());
+
+    String workflowId = null;
+    InstallmentSynchronizeDTO.ActionEnum action = installmentSynchronizeDTO.getAction();
+    switch (action) {
+      case InstallmentSynchronizeDTO.ActionEnum.I ->
+        workflowId = insertionActionMassiveDebtPositionService.handleInsertion(debtPositionSynchronizeDTO, debtPosition, massive, accessToken, operatorExternalUserId);
+      case InstallmentSynchronizeDTO.ActionEnum.M ->
+        workflowId = updateActionMassiveDebtPositionService.handleUpdate(debtPositionSynchronizeDTO, massive,  accessToken);
+      case InstallmentSynchronizeDTO.ActionEnum.A ->
+        workflowId = updateActionMassiveDebtPositionService.handleCancellation(debtPositionSynchronizeDTO, massive, accessToken);
+    }
+
+    return workflowId;
```

Comment:
do you have any plan to use workflowId before to return it?
```suggestion
    InstallmentSynchronizeDTO.ActionEnum action = installmentSynchronizeDTO.getAction();
    return switch (action) {
      case InstallmentSynchronizeDTO.ActionEnum.I ->
        insertionActionMassiveDebtPositionService.handleInsertion(debtPositionSynchronizeDTO, debtPosition, massive, accessToken, operatorExternalUserId);
      case InstallmentSynchronizeDTO.ActionEnum.M ->
        updateActionMassiveDebtPositionService.handleUpdate(debtPositionSynchronizeDTO, massive,  accessToken);
      case InstallmentSynchronizeDTO.ActionEnum.A ->
        updateActionMassiveDebtPositionService.handleCancellation(debtPositionSynchronizeDTO, massive, accessToken);
    };
```